src/brsTypes/components/RoSGNode.ts

```diff
@@ -1633,6 +1727,9 @@ export function createNodeByType(interpreter: Interpreter, type: BrsString): RoS
         // in the correct order.
         typeDefStack.push(typeDef);
         while (typeDef) {
+            // Add the current typedef to the subtypeHierarchy
+            subtypeHierarchy.set(typeDef.name!.toLowerCase(), typeDef.extends || "Node");
```

Comment:
It looks like the `subtypeHierarchy` map tends to contain lowercase keys and values — should `"Node"` be lowercase here as well?

Reply:
@sjbarag keys are lowercase, but values aren't.

The values aren't lowercase, so we can get the "true" names of the parent types (for display, logging, etc).
However, when checking "isSubType()" we must at that point perform the isLowerCase() on both the input and the value.

Does that make sense?